packages/at_client/lib/src/decryption_service/shared_key_decryption.dart

```diff
@@ -31,64 +29,76 @@ class SharedKeyDecryption implements AtKeyDecryption {
           exceptionScenario: ExceptionScenario.decryptionFailed);
     }
     String? encryptedSharedKey;
-    if (atKey.metadata != null && atKey.metadata!.pubKeyCS != null) {
+    if (atKey.metadata != null) {
       encryptedSharedKey = atKey.metadata!.sharedKeyEnc;
-      String? currentAtSignPublicKey;
-      try {
-        currentAtSignPublicKey = (await atClient
-                .getLocalSecondary()!
-                .getEncryptionPublicKey(atClient.getCurrentAtSign()!))
-            ?.trim();
-      } on KeyNotFoundException {
-        throw AtPublicKeyNotFoundException(
-            'Failed to fetch the current atSign public key - public:publickey${atClient.getCurrentAtSign()!}',
-            intent: Intent.fetchEncryptionPublicKey,
-            exceptionScenario: ExceptionScenario.localVerbExecutionFailed);
-      }
-      if (currentAtSignPublicKey != null &&
-          atKey.metadata!.pubKeyCS !=
-              EncryptionUtil.md5CheckSum(currentAtSignPublicKey)) {
-        throw AtPublicKeyChangeException(
-            'Public key has changed. Cannot decrypt shared key ${atKey.toString()}',
-            intent: Intent.fetchEncryptionPublicKey,
-            exceptionScenario: ExceptionScenario.encryptionFailed);
-      }
-    } else {
-      encryptedSharedKey = await _getEncryptedSharedKey(atKey);
     }
-    if (encryptedSharedKey == null ||
-        encryptedSharedKey.isEmpty ||
-        encryptedSharedKey == 'null') {
+    encryptedSharedKey ??= await _getEncryptedSharedKey(atKey);
+    if (encryptedSharedKey.isEmpty || encryptedSharedKey == 'null') {
       throw SharedKeyNotFoundException('shared encryption key not found',
           intent: Intent.fetchEncryptionSharedKey,
           exceptionScenario: ExceptionScenario.fetchEncryptionKeys);
     }
-    String decryptedValue = '';
+    String? currentAtSignPublicKey;
     try {
-      final decryptionResult = atClient.atChops!
+      currentAtSignPublicKey = (await _atClient
+              .getLocalSecondary()!
+              .getEncryptionPublicKey(_atClient.getCurrentAtSign()!))
+          ?.trim();
+    } on KeyNotFoundException {
+      throw AtPublicKeyNotFoundException(
+          'Failed to fetch the current atSign public key - public:publickey${_atClient.getCurrentAtSign()!}',
+          intent: Intent.fetchEncryptionPublicKey,
+          exceptionScenario: ExceptionScenario.localVerbExecutionFailed);
+    }
+    if (currentAtSignPublicKey != null &&
+        atKey.metadata != null &&
+        atKey.metadata!.pubKeyCS != null &&
+        atKey.metadata!.pubKeyCS !=
+            EncryptionUtil.md5CheckSum(currentAtSignPublicKey)) {
+      throw AtPublicKeyChangeException(
+          'Public key has changed. Cannot decrypt shared key ${atKey.toString()}',
+          intent: Intent.fetchEncryptionPublicKey,
+          exceptionScenario: ExceptionScenario.encryptionFailed);
```

Comment:
Should be ExceptionScenario.decryptionFailed?

Reply:
yes Gary. changed